linux/parser.bsh

```diff
@@ -18,6 +18,28 @@ fi
 # Functions to parse and/or validate common strings like URLs, etc.
 #**
 
+#**
+# .. envvar:: JUST_URI_REGEX
+#
+# A (best-effort) regex to parse a URL
+#
+# The following regex is based on https://tools.ietf.org/html/rfc3986#appendix-B with additional sub-expressions to split authority into userinfo, host and port
+#
+# .. warning::
+#
+#   Parsing URLs 100% correctly is HARD. This is not that solution.
+#
+# .. note::
+#    From https://stackoverflow.com/a/45977232
+#**
+
+JUST_URI_REGEX='^(([^:/?#]+):)?(//((([^:/?#]+)@)?([^:/?#]+)(:([0-9]+))?))?((/|$)([^?#]*))(\?([^#]*))?(#(.*))?$'
```

Comment:
This isn't a JUST thing, so you could rename it...

URI_REGEX
VSI_URI_REGEX
VSI_COMMON_URI_REGEX
PARSER_URI_REGEX
etc...